stock_barcodes/wizard/stock_barcodes_read_inventory.py

```diff
@@ -31,14 +31,24 @@ def _prepare_inventory_line(self):
             'prod_lot_id': self.lot_id.id,
         }
 
+    def _prepare_inventory_line_domain(self, log_scan=False):
+        """
+        Use the same domain for create or update a stock inventory line.
+        Source data is scan log record if undo or wizard model if create or
+        update one
+        """
+        record = log_scan or self
+        return [
+            ('inventory_id', '=', self.inventory_id.id),
+            ('product_id', '=', record.product_id.id),
+            ('location_id', '=', record.location_id.id),
+            ('prod_lot_id', '=', record.lot_id.id),
+        ]
+
     def _add_inventory_line(self):
         StockInventoryLine = self.env['stock.inventory.line']
-        line = first(StockInventoryLine.search([
-            ('inventory_id', '=', self.inventory_id.id),
-            ('product_id', '=', self.product_id.id),
-            ('location_id', '=', self.location_id.id),
-            ('prod_lot_id', '=', self.lot_id.id),
-        ]))
+        line = first(StockInventoryLine.search(
```

Comment:
Why use first function instead of parameter limit=1?

Reply:
Yes!!!! Fixed here https://github.com/OCA/stock-logistics-barcode/pull/195/commits/251cafa67c5fb70e9c92a38397106be31c4fe2e5